core/src/client/lwm2m_client.c

```diff
@@ -366,6 +378,37 @@ static int Lwm2mClient_Start(Options * options)
     return result;
 }
 
+static void LoadCertificateFile(char * filename)
+{
+    FILE * file = fopen(filename, "r");
+    if (file)
+    {
+        fseek(file, 0, SEEK_END);
+        long int fileSize = ftell (file);
+        if (fileSize > 0)
+        {
+            loadedClientCert = (unsigned char *)malloc(fileSize);
+            if (loadedClientCert)
+            {
+                rewind(file);
+                size_t totalRead = 0;
+                void * position = loadedClientCert;
+                size_t read;
```

Comment:
using read for a variable name might not be the best idea as it will shadow the read() function; which in this case should be okay as you aren't trying to call read() within this scope, but perhaps renaming the variable to bytesRead would be safer.